kolibri/plugins/style_guide/assets/src/views/shell/component-template.vue

```diff
@@ -0,0 +1,240 @@
+<template>
+
+  <div class="component-template">
+
+    <h1>{{ api.name }}</h1>
+
+
+    <p>{{ api.description }}</p>
+
+
+    <h2>Status</h2>
+
+    <p v-if="status === 'complete'" class="status-complete">Fully implemented</p>
+    <p v-else class="status-incomplete">Not fully implemented</p>
+
+
+    <h2 id="api">API</h2>
+
+    <h3>Import</h3>
+    <code>{{ importString }}</code>
+
+
+    <template v-if="api.props.length">
+      <h3>Props</h3>
+      <table>
+        <tr>
+          <th>Name</th>
+          <th>Type</th>
+          <th>Required</th>
+          <th>Default</th>
+          <th>Description</th>
+        </tr>
+        <tr v-for="(prop, i) in api.props" :key="i">
+          <td>{{ prop.name }}</td>
+          <td><code>{{ parsePropType(prop.value.type) }}</code></td>
+          <td><code>{{ parsePropRequired(prop.value.required) }}</code></td>
+          <td>
+            <code v-if="parsePropDefault(prop.value.type, prop.value.default)">
+              {{ parsePropDefault(prop.value.type, prop.value.default) }}
+            </code>
+            <template v-else>-</template>
+          </td>
+          <td>{{ prop.description ? prop.description : '-' }}</td>
+        </tr>
+      </table>
+    </template>
+
+
+    <template v-if="api.events.length">
+      <h3>Events</h3>
+      <table>
+        <tr>
+          <th>Name</th>
+          <th>Description</th>
+        </tr>
+        <tr v-for="(event, i) in api.events" :key="i">
+          <td>{{ event.name }}</td>
+          <td>{{ event.description ? event.description : '-' }}</td>
+        </tr>
+      </table>
+    </template>
+
+
+    <template v-if="api.slots.length">
+      <h3>Slots</h3>
+      <table>
+        <tr>
+          <th>Name</th>
+          <th>Description</th>
+        </tr>
+        <tr v-for="(slot, i) in api.slots" :key="i">
+          <td>{{ slot.name }}</td>
+          <td>{{ slot.description ? slot.description : '-' }}</td>
+        </tr>
+      </table>
+    </template>
+
+
+    <h2 id="examples">Examples</h2>
+    <vuep :template="codeExamplesTemplate"/>
+
+
+    <h2 id="guidelines">Guidelines</h2>
+    <slot name="guidelines"/>
+
+  </div>
+
+</template>
+
+
+<script>
+
+  import escodegen from 'escodegen';
+  import CamelCase from 'lodash/camelCase';
+
+  export default {
+    props: {
+      codeExamplesTemplate: {
+        type: String,
+        required: true,
+      },
+      api: {
+        type: Object,
+        required: true,
+      },
+      status: {
+        type: String,
+        default: 'incomplete',
+        validator(value) {
+          return ['complete', 'incomplete'].includes(value);
+        },
```

Comment:
Good point. It's because originally it was going to be ternary, not started, started, complete. Will address in https://github.com/learningequality/kolibri/pull/1903